### ISSUE TITLE
Pipe configs not propagating into new ember builds

### ISSUE THREAD

**Issue by Datamance** [lookit/lookit-api#220]
It looks like the mistargeting of staticfile/ember builds related to #209 is uncovering some issues we weren't seeing previously.

Here you can see the [old build.sh](https://github.com/lookit/lookit-api/blob/8df19c804cd0d46fb202de92a450d8b3633697b4/ember_build/build.sh) pulling in lots of things that were probably related to/necessary for proper pipe configuration:

```bash
sed -i "s/prepend: ''/$REPLACEMENT/g" /checkout-dir/ember-cli-build.js
sed -i "s/VideoRecorder.swf/$RECORDER_REPLACEMENT/g" /checkout-dir/lib/exp-player/addon/services/video-recorder.js

# Copy in required files
cp /environment /checkout-dir/.env
cp /VideoRecorder.swf /checkout-dir/public/
```

All that stuff is gone because when I was testing my earlier changes, the actual (fresh) experiment builds were going into a nested directory, so anything other than JSON changes weren't actually reflected in the new builds. 

We're going to fix this by adding these instructions back into our image builds.

**Comment by kimberscott** [comment]
We don't need the lines about the .swf files (I promise), but do need .env!

**Comment by Datamance** [comment]
So the `.env` copy is still there, but in the Dockerfile.

That said, I am seeing this in the browser console:

```
Timing event captured for recorderReady 
Object
eventType: "exp-video-config:recorderReady"
pipeId: ""
streamTime: null
timestamp: "2019-01-07T20:41:06.532Z"
videoId: "videoStream_5a546b08-a6ff-425a-b6d3-18cb26355a70_0-video-config_PREVIEW_DATA_DISREGARD_1546893666524_219"
__proto__: Object```

that missing `pipeId` can't be good. I'm running an image build from my console right now to see what's happening with that environment file.

**Comment by Datamance** [comment]
ok, I found the real issue. `.env` configs never actually get used - they're blank in all three environments. That lead me to look back to the [old studies/tasks.py](https://github.com/lookit/lookit-api/blob/4cbd38cc330707ea8243addf4d54c670be0bccf7/studies/tasks.py#L185) to see what I could be missing:

Indeed, I am a fool of a Took:

        ```build_command = [
            'docker',
            'run',
            '--rm',
            '-e', f'CHECKOUT_DIR={container_checkout_directory}',
            '-e', f'REPLACEMENT={re.escape(replacement_string)}',
            '-e', f'STUDY_OUTPUT_DIR={container_destination_directory}',
            '-e', f"SENTRY_DSN={os.environ.get('SENTRY_DSN_JS', None)}",
            '-e', f"PIPE_ACCOUNT_HASH={os.environ.get('PIPE_ACCOUNT_HASH', None)}",
            '-e', f"PIPE_ENVIRONMENT={os.environ.get('PIPE_ENVIRONMENT', None)}",
            '-v', f'{local_checkout_path}:/checkouts',
            '-v', f'{local_deployments_path}:/deployments',
            'ember_build'
        ]```

The container is run _**directly**_ with these environment variables.

It pains me to say this, but as of the discovery/exposure of this Pipe stuff,  I’m going to have to walk back pretty much all the previous work I did on the build pipeline and move back to a container-delegated build - there’s another reason we can’t keep it that’s not just speed, it’s how docker images store their build context (in docker history). Namely passing secrets to docker images is insecure, unless we use staged builds (the thing I wanted to introduce later when we get a chance to really refactor things).

IMO it is _extremely_ important that we keep video secure and not leave our Pipe secrets exposed anywhere. We’ll have to wait until I have some time to do all the fun staged build stuff later.

In the meantime, as an added bonus, the walk-back will also alleviate the previous build space issue that we would occasionally encounter.
